core/src/providers/auth/oAuth2ImplicitGrant.js

```diff
@@ -79,8 +79,27 @@ export class oAuth2ImplicitGrant {
     window.location.href = logouturl;
   }
 
-  renewToken() {
-    console.error('​renewToken is not implemented yet');
+  setTokenExpirationAction() {
+    const expirationCheckInterval = 5000;
+    const logoutBeforeExpirationTime = 85000000;
+
+    setInterval(() => {
+      let authData;
+      try {
+        authData = JSON.parse(localStorage.getItem('luigi.auth'));
+      } catch (e) {
+        console.warn(
+          'Error parsing authorization data. Auto-logout might not work!'
+        );
+      }
+      const tokenExpirationDate = authData.accessTokenExpirationDate;
+      const currentDate = new Date();
+
+      if (tokenExpirationDate - currentDate - logoutBeforeExpirationTime < 0) {
+        localStorage.removeItem('luigi.auth');
+        window.location = this.settings.logoutUrl + '?reason=tokenExpired';
```

Comment:
Please set the _logoutUrl_ variable to an empty string if it's not defined, otherwise it redirects to `undefined?reason=tokenExpired`.

Reply:
Same thing in the _oAuth2ImplicitGrant_.

Reply:
both fixed